src/rust/ensogl/lib/components/src/list_view/entry.rs

```diff
@@ -18,148 +20,261 @@ use ensogl_theme;
 pub const PADDING:f32 = 14.0;
 /// The overall entry's height (including padding).
 pub const HEIGHT:f32 = 30.0;
-/// The text size of entry's labe.
-pub const LABEL_SIZE:f32 = 12.0;
-/// The size in pixels of icons inside entries.
-pub const ICON_SIZE:f32 = 0.0; // TODO[ao] restore when we create icons for the searcher.
-/// The gap between icon and label.
-pub const ICON_LABEL_GAP:f32 = 0.0; // TODO[ao] restore when we create icons for the searcher.
 
 
 
-// ===================
-// === Entry Model ===
-// ===================
+// ==================================
+// === Type Aliases and Reexports ===
+// ==================================
 
 /// Entry id. 0 is the first entry in component.
 pub type Id = usize;
 
-/// A model on which the view bases.
-#[allow(missing_docs)]
-#[derive(Clone,Debug,Default)]
-pub struct Model {
-    pub label       : String,
-    pub highlighted : Vec<text::Range<text::Bytes>>,
-    pub icon        : Option<display::object::Any>,
+pub use list::List;
+
+
+
+// =============
+// === Trait ===
+// =============
+
+/// An object which can be entry in [`crate::ListView`] component.
+///
+/// The entries should not assume any padding - it will be granted by ListView itself. The Display
+/// Object position of this component is docked to the middle of left entry's boundary. It differs
+/// from usual behaviour of EnsoGl components, but makes the entries alignment much simpler.
+///
+/// The ListView component does not create Entry object for each entry provided, only those visible.
+/// During scrolling, the instantiated objects will be reused: they position will be changed and
+/// they will be updated using `update` method.
```

Comment:
this is very important - I would write here in the ocs that this trait abstracts over model and its updating in order to support such and such a case (which is required for performance reasons)